flocx_market/tests/integration/models/test_offer.py

```diff
@@ -0,0 +1,21 @@
+from flocx_market.db.sqlalchemy.offer_api import OfferApi
+from flocx_market.tests.base_test import BaseTest
+import datetime
+
+now = datetime.datetime.utcnow()
+
+
+class TestOffer(BaseTest):
+
+    def test_crud(self):
+        with self.app_context():
+            offer = OfferApi('b9752cc0-9bed-4f1c-8917-12ade7a6fdbe',
+                             'Alice1992',
+                             now, 'available', 'Dell380', now,
+                             now,
+                             '{"new attribute XYZ": "This is just a sample list of free-form attributes used for describing a server.","cpu_type": "Intel Xeon","cores": 16,"ram_gb": 512,"storage_type": "samsung SSD", "storage_size_gb": 204}',
```

Comment:
This looks like a problem: The offer should accept a *dictionary* for the `server_config` field. The serialization to JSON for storage in the datbase (and un-serialization when a value is loaded) is an implementation detail that should not be visible to calling code.